packages/babel-plugin-transform-es2015-parameters/src/rest.js

```diff
@@ -103,33 +102,47 @@ let memberExpressionOptimisationVisitor = {
     }
   }
 };
-
 function hasRest(node) {
   return t.isRestElement(node.params[node.params.length - 1]);
 }
 
+function optimiseIndexGetter(path, argsId, offset) {
+  path.parentPath.replaceWith(loadRest({
+    ARGUMENTS: argsId,
+    INDEX: t.numericLiteral(path.parent.property.value + offset)
+  }));
+}
+
+function optimiseLengthGetter(path, argsLengthExpression, argsId, offset) {
+  if (offset) {
+    path.parentPath.replaceWith(
+      t.binaryExpression(
+        "-",
+        argsLengthExpression,
+        t.numericLiteral(offset),
+      )
+    );
```

Comment:
I decided not to use `templates` to avoid the overhead and because this is a pretty simple expression.
Let me know if using that would be preferred.